eth_common/eth_types.nim

```diff
@@ -121,14 +121,33 @@ else:
 # Rlp serialization:
 #
 
+func trailingNonZeros(val: openarray[byte]): int =
+  for i in countdown(val.len - 1, 0):
+    if val[i] == 0:
+      return val.len - i - 1
+
+  return val.len
```

Comment:
You will have a corner case for negative int, for example for a uint16:

10000000 11111111 --> should we take that as 1 trailing non zero, ignoring the sign or 2?

Note that there is a "countLeadingZeroBits" in Stint for uint but I don't think it maps cleanly to bytes (https://github.com/status-im/nim-stint/blob/master/stint/uint_public.nim)

Reply:
Good catch. The RLP spec is defined only over unsigned integers, so I'll change the signature of the procs to handle only UInt values.